source/DEVS-Network/DEVSNetworkNode.class.st

```diff
@@ -0,0 +1,51 @@
+Class {
+	#name : 'DEVSNetworkNode',
+	#superclass : 'DEVSAtomicComponent',
+	#instVars : [
+		'state',
+		'messages',
+		'routingTable'
+	],
+	#category : 'DEVS-Network',
+	#package : 'DEVS-Network'
+}
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> externalTransition [
+
+	| msg |
+	msg := self peekFrom: #in.
+	messages add: msg.
+	state := #processing 
+]
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> initialize [ 
+	super initialize.
+	self addInputPortNamed: #in.
+	self addOutputPortNamed: #out.
+	state := #idle.
+	messages := OrderedCollection new.
+	routingTable := Dictionary new
```

Comment:
Purpose of routing table? I forgot network stuff, so I don't know whose responsibility is routing table... and maybe you have some follow-up intention here.  But at this moment, traffic is directed via port couplings?

Reply:
I tried to do fully automatic meaning that a msg is looked up in the routingTable and added to the next components in port. But as I wrote this shortcuts the adding to the eventList and didn't work. That's why I got back to making one output port for each address

Reply:
To be replaced by dns dictionary you showed me.